docs/user/adding-new-metrics.md

```diff
@@ -1,97 +1,216 @@
 
-### Naming things
+While lifetimes are important to understand for all metric types, they are particularly important for the metric types that record single values and don't aggregate on the client (`boolean`, `string`, `labeled_string`, `string_list`, `datetime` and `uuid`), since these metrics will send the "last known" value and missing the earlier values could be a form of unintended data loss.
+
+#### A lifetime example
+
+Let's work through an example to see how these lifetimes play out in practice. Let's suppose we have a user preference, "turbo mode", which defaults to `false`, but the user can turn it to `true` at any time.  We want to know when this flag is `true` so we can measure its affect on other metrics in the same ping.  In the following diagram, we look at a time period that sends 4 pings across two separate runs of the application. We assume here, that like Glean's built-in [metrics ping](pings/metrics.html), the developer writing the metric isn't in control of when the ping is submitted. 
+
+In this diagram, the ping "lifetimes" are represented as rectangles, but the moment the ping is "submitted" is represented by its right edge. The user changes the "turbo mode" setting from `false` to `true` at 3 seconds.
+  
+```mermaid
+gantt
+  axisFormat %S
+  title Metric lifetimes
+  dateFormat  s
+
+  section Metrics<br/>pings
+    Ping 1 :active, p1, 0, 2s
+    Ping 2 :active, p2, 2, 3s
+    Ping 3 :active, p3, 5, 10s
+    Ping 4 :active, p4, 15, 5s
+    Ping 5 :active, p5, 20, 5s
+
+  section Process
+	Run 1     :done, a1, 0, 10s
+	Run 2     :done, a2, 15, 10s
 
-Metric identifiers (the combination of a metric's category and name) must be unique across all metrics that are sent by an application.
+  section Interaction
+    ↓ Turns on turbo mode : 3
+
+  section Ping<br/>(change)
+	True      :3, 2s
+
+  section Ping<br/>(init and change)
+    False     :0, 2s
+    True      :3, 2s
+    True      :15, 5s
+
+  section Application<br/>(change)
+    True      :3, 12s  
+
+  section Application<br/>(init and change)
+    False     :0, 3s
+    True      :3, 12s
+    True      :15, 10s
+
+  section User<br/>(change)
+    True      :3, 22s
+```
+
+- **Ping (change)**: The lifetime is set to `ping` and the value is recorded into the metric when it changes. The value isn't included in Ping 1, because Glean doesn't know about it yet.  It is included in the first ping after being recorded (Ping 2), which causes it to be cleared and it is not included in any subsequent pings.
+
+- **Ping (init and change)**: The lifetime is set to `ping` and the value is recorded at application initialization and when it changes. The default value is included in Ping 1, and the changed value is included in Ping 2, which causes it to be cleared.  It therefore misses Ping 3, but when the application is started, it is recorded again and it is included in Ping 4.  However, this causes it to be cleared again and it is not in Ping 5.
```

Comment:
Is "Ping" here meaning a custom ping? Or the 'metrics' ping? I think we should disambiguate as it gets confusing: 'ping' is a lifetime and a ping name

Reply:
It's ping lifetime.  Will disambiguate.